inc/classes/class-post-modifier_class.php

```diff
@@ -28,9 +29,18 @@ public function setSettings( $settings ) {
 	}
 
 	public function updateSetting( $name, $value ) {
-		$settings[ $name ] = $value;
+		$this->settings[ $name ] = $value;
 		update_option( $name, $value );
 	}
+
+	public function updateLogoSrcs( $device, $value ) {
+		$this->settings['logo_srcs'][Constants::device_indexes($device)]['src'] = $value;
```

Comment:
try https://www.php.net/manual/en/function.array-search.php called on `$this->settings['logo_srcs']` for the `$device` key ( and you can get rid of `device_indexes` then )